core-api/src/telemetry/metrics.rs

```diff
@@ -0,0 +1,263 @@
+use std::{fmt::Debug, sync::Arc};
+
+/// Implementors of this trait are expected to be defined in each language's bridge.
+/// The implementor is responsible for the allocation/instantiation of new metric meters which
+/// Core has requested.
+pub trait CoreMeter: Send + Sync + Debug {
+    fn new_attributes(&self, attribs: MetricsAttributesOptions) -> MetricAttributes;
+    fn counter(&self, name: &str) -> Arc<dyn Counter>;
```

Comment:
Are there any perf concerns with the cost of `dyn` for metrics? Is this dynamically dispatched for each metrics call?

Reply:
See other answer. Kinda need to to keep things sane. That's not the overhead I'm worried about so much as attribute passing.